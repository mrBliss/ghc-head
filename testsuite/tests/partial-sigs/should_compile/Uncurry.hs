module Uncurry where

unc :: (_ -> _ -> _) -> (_, _) -> _
unc = uncurry


foo :: Char -> Bool -> Int
foo = undefined

test :: ([Bool], Int, (Bool, Char))
test = (unc (:) (True, []), unc foo ('c', True), unc (,) (True, 'c'))
