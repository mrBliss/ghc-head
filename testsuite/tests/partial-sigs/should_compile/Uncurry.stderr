TYPE SIGNATURES
  unc ::
    forall tw_ tw_1 tw_2. (tw_ -> tw_1 -> tw_2) -> (tw_, tw_1) -> tw_2
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
